array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,22 @@
+/* exported compact */
+// function compact(array) {
+//   var newArray = [];
+//   for (var i = 0; i < array.length; i++) {
+//     if (array[i] !== false && array[i] !== '' && array[i] !== -0 && array[i] !== 0 && array[i] !== undefined && array[i] !== null) {
+//       if (array[i] === array[i]) {
+//         newArray.push(array[i]);
+//       };
+//     };
+//   };
+//   return newArray;
+// };
+
+function compact(array) {
+  var newArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (array[i] > 0 || array[i] < 0 || array[i] === true || (array[i] !== '' && typeof (array[i]) === typeof ('')) || (typeof (array[i]) === typeof ([]) && array[i] !== null)) {
```

Comment:
You can check if a value is truthy by just passing the value directly into the condition of an if statement, like so: `if(array[i])`